### PR TITLE
[4] article field List on webservices

### PR DESCRIPTION
**Pull request by alikon**
Pull Request for Issue #42200 .

### Summary of Changes
fixed when non Multiple choice 

### Testing Instructions
Add a new Field of type List to your Article. Select a value and save the article.

Get the Article over the Webservices API

GET /api/index.php/v1/content/articles/{{yourArticleId}}

### Actual result BEFORE applying this Pull Request
`"fieldlist": [0]`

### Expected result AFTER applying this Pull Request

`"fieldlist": {
                "2": "Two"
            }`

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [ ] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [ ] No documentation changes for manual.joomla.org needed

### CODE REVIEW

File: plugins/fields/list/src/Extension/ListPlugin.php [hunk 1]
```diff
@@ -47,10 +47,14 @@ public function onCustomFieldsBeforePrepareField($context, $item, $field)
...
-        if (is_array($field->value)) {
+        if (\is_array($field->value)) {
             foreach ($field->value as $value) {
                 $field->apivalue[$value] = $options[$value];
             }
+        } else {
+            if (!empty($field->value)) {
+                $field->apivalue = [$field->value => $options[$field->value]];
+            }
         }
```
Comment (Quy) [inline plugins/fields/list/src/Extension/ListPlugin.php]:
```suggestion
        } elseif (!empty($field->value)) {
            $field->apivalue[$field->value] = $options[$field->value];
        }
```

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 962fbefc2f7699ded1d6ec34dba38618f106863c<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42206">issues.joomla.org/tracker/joomla-cms/42206</a>.</sub>

**Comment by toivo** [comment]
I have tested this item :white_check_mark: successfully on 962fbefc2f7699ded1d6ec34dba38618f106863c<br /><br />Tested successfully in Joomla 4.4.1-dev of 21 October in Wampserver using PHP 8.2.11 and Postman Desktop Agent<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42206">issues.joomla.org/tracker/joomla-cms/42206</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42206">issues.joomla.org/tracker/joomla-cms/42206</a>.</sub>

**Pull request merged**